src/sidebar/components/menu-keyboard-navigation.js

```diff
@@ -42,6 +57,7 @@ export default function MenuKeyboardNavigation({
 
   const onKeyDown = event => {
     const menuItems = Array.from(
+      /** @ts-ignore - assume current is not null */
```

Comment:
It would not allow a cast here... no "overlap" between null and HTMLElement prevented that. I think what it wants is a simple condition on `current`, but I think we can safely assert `current` is not null here so I added the `ignore`

Reply:
Ah - subtle problem here. You're missing the parens around the `null`  in the `useRef` call. If you add those, you don't need the `@ts-ignore` here. You can also remove the `Ref` import at the top of this file now.

In other words:

```diff
- const menuRef = useRef(/** @type {HTMLElement|null} */ null);
+ const menuRef = useRef(/** @type {HTMLElement|null} */ (null));
```

I appreciate it is non-obvious here that the "null-ness" is ignored whenever you access `menuRef.current`, but if we can accept this quirk for now and use it consistently, it will be easy to update everywhere in future if we find something better.

As an aside, in TS syntax (ie. not constrained to JS syntax) it is somewhat easier as you can do something like:

```ts
const aRef = useRef<HTMLElement>();
aRef.current!.querySelector(...) // `!` means "trust me, I know what I'm doing"
```

Reply:
ya, TS can handle that a bit more gracefully, 

**facepalm** on the parens